components/mpas-ocean/src/shared/mpas_ocn_scaled_sfwf.F

```diff
@@ -0,0 +1,557 @@
+!
+!  ocn_scaled_sfwf
+!> \brief MPAS ocean scale data freshwater fluxes
+!> \author Shixuan Zhang 
+!> \date   May 2025
+!> \details
+!>  Computes a precipitation factor to multiply the fresh water flux
+!>  due to precipitation uniformly to ensure a balance of fresh water
+!>  at the ocean surface.
+!>  Designed following precip_adjustment subroutine located at:
+!>    https://github.com/ESCOMP/POP2-CESM/blob/master/source/forcing_sfwf.F90
+!
+!-----------------------------------------------------------------------
+
+module ocn_scaled_sfwf
+
+    use mpas_kind_types
+    use mpas_derived_types
+    use mpas_global_sum_mod
+    use mpas_timekeeping
+    use mpas_timer
+
+    use ocn_config
+    use ocn_mesh
+
+    use shr_kind_mod, only: & 
+        SHR_KIND_R8
+
+    use shr_const_mod
+
+    use ocn_constants, only: & 
+        rho_fw, & 
+        rho_sw, &
+        ocn_ref_salinity
+
+    implicit none
+    private
+    save
+
+    !--------------------------------------------------------------------
+    !
+    ! Public parameters
+    !
+    !--------------------------------------------------------------------
+
+    !--------------------------------------------------------------------
+    !
+    ! Public member functions
+    !
+    !--------------------------------------------------------------------
+
+    public :: ocn_init_scaled_sfwf, &
+              ocn_update_scaling_factor
+
+    !--------------------------------------------------------------------
+    !
+    ! Private module variables
+    !
+    !--------------------------------------------------------------------
+
+    logical :: scaledSFWFOn
+    character (len=*), parameter :: alarmID = 'scaledSFWFUpdateAlarm'
+
+    real (kind=RKIND) :: earthAreaE3SM ! surface area of earth in coupler
+
+    real(kind=RKIND) :: rhow, areaCellGlobal, volumeCellGlobal
+
+!***********************************************************************
+
+contains
+
+
+!***********************************************************************
+!
+!  routine ocn_init_scaled_sfwf
+!
+!> \brief   Initialize scaling of data freshwater fluxes
+!> \author Shixuan Zhang 
+!> \date   May 2025
+!> \details
+!>  Set alarms needed to compute yearly and running means of water fluxes 
+!
+!-----------------------------------------------------------------------
+
+    subroutine ocn_init_scaled_sfwf(domain)!{{{
+
+        type (domain_type), intent(inout) :: domain
+
+        ! Alarm variables
+        type (mpas_Time_Type) :: alarmTime
+        type (mpas_TimeInterval_type) :: alarmTimeStep
+
+        integer :: err_tmp
+
+        ! taking PI from SHR_CONST_PI in share/util/shr_const_mod.F90 to match coupler
+        real (kind=RKIND), parameter :: piE3SM = SHR_CONST_PI
+        ! taking earth radius from SHR_CONST_REARTH in share/util/shr_const_mod.F90 to match coupler
+        real (kind=RKIND), parameter :: earthRadiusE3SM = SHR_CONST_REARTH ! radius of earth, m
```

Comment:
```suggestion
```
See comment about getting these from ocn_constants

Reply:
I see that this was copied from the conservation AM https://github.com/E3SM-Project/E3SM/blob/7ac7d34707ca22c2555ca86a79a5c07b5a54fc4d/components/mpas-ocean/src/analysis_members/mpas_ocn_conservation_check.F#L163-L170. I still think getting it from `ocn_constants` is better though.